python/cudf/cudf/core/frame.py

```diff
@@ -1627,10 +1614,15 @@ def rank(
                 "na_option must be one of 'keep', 'top', or 'bottom'"
             )
 
-        # TODO code for selecting numeric columns
         source = self
         if numeric_only:
-            warnings.warn("numeric_only=True is not implemented yet")
+            numeric_cols = filter(
+                lambda name: is_numerical_dtype(self._data[name]),
+                self._data.names,
+            )
```

Comment:
```suggestion
            numeric_cols = (
                name for name in self._data.names if is_numerical_dtype(self._data[name])
            )
```

A list comprehension is possibly [more Pythonic](https://www.artima.com/weblogs/viewpost.jsp?thread=98196) than filter+lambda?

Reply:
Cool! Thanks for pointing me to the reading.